packages/popup/src/pages/CurrentSeed/index.js

```diff
@@ -10,18 +10,42 @@ const useStore = create(
     // value
     hdGroup: [],
     creatingAccount: false,
-    accountCreatationError: null,
+    accountCreationError: null,
+    accountNameError: null,
     accountName: '',
     selectedGroup: null,
 
     // hook
+    groupBeforeSet: ({groupData, ...others}) => ({
+      groupData: groupData.map((g, index) => {
+        g.index = index
+        return g
+      }),
+      ...others,
+    }),
     groupAfterSet: ({groupData}) => {
-      set({hdGroup: groupData.filter(({vault: {type}}) => type === 'hd')})
-      if (!get().selectedGroup) set({selectedGroup: get().group.groupData[0]})
+      const hdGroup = groupData.filter(g => g?.vault?.type === 'hd')
+      set({hdGroup})
+      if (!get().selectedGroup) set({selectedGroup: hdGroup[0]})
+      if (!get().accountName && hdGroup[0]) {
+        const index = groupData.filter(g => g.eid === hdGroup[0].eid)[0].index
```

Comment:
we assign the group index in beforeSet and the hdGroup is in afterSet, the group in hdGroup should have the index